### ISSUE TITLE
Create a plugin for this class

### ISSUE THREAD

**Issue by GaryJones** [radishconcepts/WordPress-GitHub-Plugin-Updater#55]
Apologies if this has been suggested before - I looked through the issues but couldn't spot anything.

Getting X developers to include different versions of the class (as it it improves) in Y plugins seems to be taking the wrong approach. I'd rather see the class turned into something more like ACF (though I've not used ACF myself) - that is, an individual plugin that enables the other plugins to use it if it exists. This updater plugin could then be improved, and so long as the API between the other plugins and the class plugins stays the same, nothing should break.

To put it another way, WPGHPU becomes a plugin which adds another repository for WordPress, and other plugins can tap into that, if it exists.

Thoughts?
